libexec/rtld-elf/mips/Makefile.inc

```diff
@@ -4,3 +4,29 @@ CFLAGS.reloc.c+=-fno-jump-tables
 
 # XXX-MAYBE-OBE: Added in 46fa9656e6167f41a3de4dc9faed3722f4124440
 RTLD_ENTRY=	rtld_start
+
+.if ${MACHINE_ABI:Mpurecap}
+CLEANFILES+=	cheri_plt_code.dump cheri_plt_code.text cheri_plt_code.inc
+
+cheri_plt_code.dump: cheri_plt_code.o
+	${OBJDUMP:Uobjdump} -triple=mips64-unknown-freebsd -d -x -s cheri_plt_code.o > ${.TARGET}
+
+cheri_plt_code.text: cheri_plt_code.o
+	${OBJCOPY} -O binary --only-section=.text cheri_plt_code.o ${.TARGET}
+
+# Generate the code for the PLT stubs as a single C array using objcopy+xxd/file2c
+cheri_plt_code.inc: cheri_plt_code.text
+# ${XXD:U/usr/local/bin/xxd} -i -c4 < cheri_plt_code.text > ${.TARGET}
```

Comment:
As a followup, it might be nice to at least drop the XXD line.  (I know this is just the existing bits.)  I also wonder if .incbin would be any better?  Or alternatively, could the function itself be a data type and placed in .rodata instead of in .text and then you just reference the symbol directly similar to what we do for sigcode in the kernel?

Reply:
I assume it's done the way it is so the `sizeof` for the `memcpy` is known at compile time, but yeah the sigcode way would be a much nicer way of doing it because the current way is rather... interesting. I don't know how much it would hurt performance, but likely not very much, this should be on a very slow path already (hence why lazy binding is a thing...). Using `.incbin` wouldn't achieve anything, we're converting in the other direction to what `.incbin` is for.

Reply:
At the time it seemed like the easiest way to write assembly rather than writing raw instruction encodings and having the size of the array as a constant. But if there's a better way of doing it I'd be happy to change it.